invenio_app_rdm/theme/views/records.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019-2020 CERN.
+# Copyright (C) 2019-2020 Northwestern University.
+# Copyright (C)      2021 TU Wien.
+#
+# Invenio App RDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Routes for record-related pages provided by Invenio-App-RDM."""
+
+from flask import abort, g, render_template, request
+from invenio_files_rest.views import ObjectResource
+from invenio_rdm_records.proxies import current_rdm_records
+
+from ..utils import obj_or_import_string, previewer_record_file_factory
+
+
+def register_records_ui_routes(app, blueprint):
+    """Dynamically registers routes (allows us to rely on config)."""
+
+    @blueprint.route(
+        app.config.get(
+            "APP_RDM_RECORDS_EXPORT_URL",
+            "/records/<pid_value>/export/<export_format>",
+        )
+    )
+    def export_record(pid_value, export_format="json", **kwargs):
+        """Record export-as landing page."""
+        template = kwargs.get(
+            "template", "invenio_app_rdm/landing_page/export_page.html"
+        )
+
+        resource = current_rdm_records.records_resource
+        service = current_rdm_records.records_service
+        links_config = resource.config.links_config
+
+        record = service.read(
+            id_=pid_value, identity=g.identity, links_config=links_config
+        )._record
```

Comment:
```suggestion
        record = service.read(
            id_=pid_value, identity=g.identity, links_config=links_config
        ).to_dict()
```
This should be enough to get a serializable result. Access to `_record` was probably done in another view, but we shouldn't do it since it's private.

Reply:
I got the `record._record` out, but had to tweak a couple things in the Jinja template to get it running.
From what I see there's no problem, but maybe there's stuff breaking somewhere under the hood that I don't see:

* had to throw out the `record.data|dereference_record` in `export_page.html`, because `record.data` (dict) doesn't have a `relations` attribute (https://github.com/inveniosoftware/invenio-app-rdm/pull/548/commits/423752e444722d6c85bc584bfa76891bfec472f4#diff-4f71f75b58c44d01d2135bfec1a989b083a16321f4833c7c997c4e1e1e75ff2eL17)
* had to comment out the access-right icon, because the `record.ui` doesn't have an `access_right` property when done this way (https://github.com/inveniosoftware/invenio-app-rdm/pull/548/commits/423752e444722d6c85bc584bfa76891bfec472f4#diff-4f71f75b58c44d01d2135bfec1a989b083a16321f4833c7c997c4e1e1e75ff2eR36)
* to preserve compatibility with existing `render_template` calls, I added a code-path that handles the `record` being specified as `RDMRecord`, or as `RecordItem` together with separate `pid_value` argument (https://github.com/inveniosoftware/invenio-app-rdm/pull/548/commits/423752e444722d6c85bc584bfa76891bfec472f4#diff-124d51df0ed3415b56175f047c630953e5df5b5e312406e74f7cb9c121493736L59)

Reply:
Thanks for highlighting these issues It's fine for now Those are all points that we need to take into account when improving that response flow / representation. Personally, I am not happy yet with the squishiness of some abstractions and we will be making improvements.